src/types/bytes.rs

```diff
@@ -203,8 +203,8 @@ impl<I: SliceIndex<[u8]>> Index<I> for Bound<'_, PyBytes> {
 /// If the source object is a `bytes` object, the `Cow` will be borrowed and
 /// pointing into the source object, and no copying or heap allocations will happen.
 /// If it is a `bytearray`, its contents will be copied to an owned `Cow`.
-impl<'source> FromPyObject<'source> for Cow<'source, [u8]> {
-    fn extract(ob: &'source PyAny) -> PyResult<Self> {
+impl<'py> FromPyObject<'py> for Cow<'py, [u8]> {
+    fn extract(ob: &'py PyAny) -> PyResult<Self> {
```

Comment:
Was not moving this to `extract_bound` intentional?

Reply:
Yes, there's a bit of complexity around `Cow`, `&str` and `&[u8]` because they need to depend on the lifetime of the input `&Bound`.

For 0.21 I don't want to touch them, they will just have to continue with the pool I think. In 0.22 or 0.23 we may have to remove their `FromPyObject` implementations and implement `PyFunctionArgument` for them. Or we keep the pool for a little longer as an internal only thing just for these three implementations.

The right long term solution will again be adding the second lifetime to `FromPyObject` so we can implement these implementations correctly.

Reply:
Uff, so this means that 0.21 will not be fully compatible with `gevent` then?

I wonder if having something like `Backed<T>` where `T` is separate Rust type with inline storage (e.g. `&[u8]`), but backed by a Python pointer would be able to solve this without adding a lifetime to the schema at the cost of a reference count bump.

Reply:
I've wondered a little about some kind of `Backed<T>` type too, but I haven't explored the practicalities of how that would actually be implemented. I sort of bucketed that as a future feature only because it's 

> Uff, so this means that 0.21 will not be fully compatible with gevent then?

One option is to gate these `FromPyObject` implementations on the `gil-refs` feature. That would be a normal "additive" feature. Without the feature enabled we implement `PyFunctionArgument` instead. I will add that as a bullet in the tracking issue and potentially will push it as a draft later.

(At least then if users follow our migration steps their code wouldn't break from this until they disable the `gil-refs` feature.)